kiva/agg/src/kiva_graphics_context.h

```diff
@@ -1184,12 +1184,16 @@ namespace kiva
             line_color.a *= this->state.line_width;
             m.line_color(line_color);
 
-            double tx, ty;
-            get_translation(ctm, &tx, &ty);
+            double mx, my, sx, sy;
+            get_scale(ctm, &sx, &sy);
 
             for(int i = 0; i < Npts*2; i+=2)
             {
-                m.marker((int)(pts[i]+tx), int(pts[i+1]+ty), size, type);
+                mx = pts[i];
+                my = pts[i+1];
+                ctm.transform(&mx, &my);
+                // XXX: Assuming scale is uniform in both directions
+                m.marker(int(mx), int(my), size * sx, type);
```

Comment:
I am not entirely sure how this change works.

Previously, we were storing the translation in x and y as `tx` and `ty` - and then adding markers at `pts[i]+tx` and `pts[i+1]+ty`.

Now, we're storing the scale information in `sx` and `sy`, transforming the context manager by `pts[i]` and `pts[i+1]` and then also add markers at `pts[i]` and `pts[i+1]`?

Reply:
`ctm` is the "current transformation matrix". We're applying it to `(mx, my)`, which the position of the current marker. Then we're calling the `marker` method on the marker renderer to render a marker at the position `(mx, my)` and scaling its size by `sx`.